### PR TITLE
Bulk update direct `go` dependencies

### PR DESCRIPTION
**Pull request by jentfoo**
This updates all direct go dependencies (ignoring examples and excluding the ones ignored by Dependabot).

The broad update will hopefully simplify the Dependabot resolutions, letting our automated updates resume.

### CODE REVIEW

File: examples/access-plugin-minimal/go.mod
Comment (codingllama) [inline examples/access-plugin-minimal/go.mod:1]: I'm not sure it's a good idea to touch examples, they aren't exercised by CI.

File: go.mod [hunk 1]
```diff
@@ -37,36 +37,36 @@ require (
 	github.com/alecthomas/kingpin/v2 v2.4.0 // replaced
 	github.com/alicebob/miniredis/v2 v2.32.1
 	github.com/andybalholm/brotli v1.1.0
-	github.com/aquasecurity/libbpfgo v0.5.1-libbpf-1.2
+	github.com/aquasecurity/libbpfgo v0.6.0-libbpf-1.3
```
Comment (codingllama) [inline go.mod:40]: Careful here. Better move this one to its own PR, it usually wants updates to libbpf itself.

File: go.mod [hunk 2]
```diff
@@ -86,10 +86,10 @@ require (
 	github.com/go-jose/go-jose/v3 v3.0.3
 	github.com/go-ldap/ldap/v3 v3.4.6
 	github.com/go-logr/logr v1.4.1
-	github.com/go-mysql-org/go-mysql v1.5.0 // replaced
+	github.com/go-mysql-org/go-mysql v1.7.0 // replaced
```
Comment (codingllama) [inline go.mod:89]: Let's not touch this one, it's replaced. It's also in our ignore list.

https://github.com/gravitational/teleport/blob/master/.github/dependabot.yml#L9

File: go.mod [hunk 3]
```diff
@@ -98,39 +98,39 @@ require (
...
 	github.com/gorilla/websocket v1.5.1
-	github.com/gravitational/license v0.0.0-20231228155916-928ed9ac0335
-	github.com/gravitational/oxy v0.0.0-20221029012416-9fbf4c444680
+	github.com/gravitational/license v0.0.0-20240313232707-8312e719d624
+	github.com/gravitational/oxy v0.0.0-20231219172753-f855322f2a6c
 	github.com/gravitational/roundtrip v1.0.2
 	github.com/gravitational/teleport/api v0.0.0
 	github.com/gravitational/trace v1.4.0
-	github.com/gravitational/ttlmap v0.0.0-20171116003245-91fd36b9004c
+	github.com/gravitational/ttlmap v0.0.0-20200608174355-7e347b203ef0
```
Comment (codingllama) [inline go.mod:114]: This one is in our ignore list, it'll cause problems for sure.

File: go.mod
Comment (codingllama) [inline go.mod:1]: Have a look at our dependabot ignores: https://github.com/gravitational/teleport/blob/master/.github/dependabot.yml#L9. You've hit a bunch of those, they are sure to cause problems.

File: examples/access-plugin-minimal/go.mod
Comment (jentfoo) [reply examples/access-plugin-minimal/go.mod:1]: It would be good to update them, we need to figure out how to be able to validate them

File: go.mod
Comment (jentfoo) [reply go.mod:1]: Thank you, I knew this would be too broad, but I wanted to see what happened.  I will go through the individual updates we want to pull back in an update and mark the PR as ready after CI is looking good.

File: examples/access-plugin-minimal/go.mod
Comment (jentfoo) [reply examples/access-plugin-minimal/go.mod:1]: I can split it into another PR :+1:

File: go.mod [hunk 4]
```diff
@@ -98,34 +98,34 @@ require (
...
 	github.com/google/renameio/v2 v2.0.0
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
 	github.com/google/uuid v1.6.0
-	github.com/googleapis/gax-go/v2 v2.12.2
+	github.com/googleapis/gax-go/v2 v2.12.3
 	github.com/gorilla/websocket v1.5.1
-	github.com/gravitational/license v0.0.0-20231228155916-928ed9ac0335
-	github.com/gravitational/oxy v0.0.0-20221029012416-9fbf4c444680
+	github.com/gravitational/license v0.0.0-20240313232707-8312e719d624
+	github.com/gravitational/oxy v0.0.0-20231219172753-f855322f2a6c
```
Comment (codingllama) [inline go.mod:110]: Big jump here. I wonder why this was untouched until now.

**Review by codingllama — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (jentfoo) [reply go.mod:110]: I am not sure, I don't see it ever being ignored in the history

Comment (codingllama) [reply go.mod:110]: CI is green so it's probably OK, but still it's weird. Maybe someone else has an idea.

**Review by rosstimothy — APPROVED** [review]
